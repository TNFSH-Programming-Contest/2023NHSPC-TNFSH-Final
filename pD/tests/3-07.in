10 158
806725667 -384028278 -135278234 357004110 -6064028 376224708 33711498 420938754 211567687 320564227 412216440 14789045 871041264 298946618 21287461 -349166992 -454077511 811182033 942609178 565888618 268255667 337857471 602815967 149134756 574277090 495584852 -23874966 169750191 -71825053 -491867712 466906765 232333302 432002762 721796879 -207669133 719950366 795919665 406014241 453750434 -170784178 -209578004 -209862323 41306079 22999128 -299421242 240942511 649593541 844451168 -54599638 232422073 -12252269 -6869307 462214205 412666966 -353315135 -265955409 185415176 746532839 977426263 413708333 758026901 -41134606 -12084641 -211523987 -153973857 -384591266 716832865 223424808 828177338 857649179 -211316943 633960 -11646247 810133999 91607587 296689424 504592942 617644658 -487301349 659897622 -63041178 522400551 55563814 -134738014 465875172 -366070541 -464182955 -321753880 754519721 486119314 997838400 470945257 224313081 365011659 609020090 301709596 951818339 800418464 -345326610 -82230375 384289292 -439237063 193356140 -303891449 812440124 560217546 -492468383 327573614 233329392 428168971 -11571147 126409738 -273620517 936459520 618589596 639114435 -306053074 -315661830 772369928 103550328 139720150 670984129 947206767 657511669 139394693 -230273330 -136245574 -439135655 989364978 70862583 -498561584 350168773 42859877 -80250269 492185498 363028783 454435400 -230503129 565403918 164300116 441367281 -99590039 -113702930 785157401 909830039 -311263289 782747276 -384982870 902990670 426190091 757844691 80116556 788109372 -158514567 -219538550 -264906525 500900575 -87622304
182464460 443740117 555380053 -480877840 764291219 689828504 572368960 353601222 -15631567 774059323 -284361091 811019324 538186239 -464069755 180052351 979502079 21231113 822200245 -363714678 311915328 -465082493 493801060 144851062 -290895801 -427784336 641780819 -372927082 -410125275 -12290632 109061043 497510376 884824361 200645106 502080796 991690073 183921500 105245412 107507076 49936644 652203868 258387484 648363111 -141113509 15149367 -413102362 -425768507 471398157 594965342 514002119 666986605 695882586 79027624 633072538 983948712 789596370 815333886 358034694 227318336 584099761 -335590697 -116328842 960736790 939015110 -221595422 -276954207 559982162 138095009 86310332 890833292 373460717 728638265 -56589855 943996355 -289643845 72611714 913023180 48664392 643431009 200104836 -401574594 -125159720 212345889 321361604 780672575 859229333 37225356 213701596 -206108688 164745803 426382295 44953535 -241099229 -426976081 221822250 890399822 -305431113 243252472 -111613255 693517880 314134061 -262550092 -329868759 77050633 -92996795 168227938 221409942 -302976887 876631852 354547459 778257711 115523331 53851697 855432975 463728802 -402206607 24318875 905854847 661275575 884860229 466903041 152506172 602706588 -279979660 61944134 526107050 -9083156 386983726 417444122 288138918 240791667 379345323 -352463655 126206666 -99490589 379662728 -321760338 -215712259 591004340 -365889521 207156136 52344898 -407914180 347831576 204168865 -491380423 -199950151 891491387 779255335 -105104957 -23372011 395875947 881731956 51184713 700259352 71931738 -17062121 310808349 535179514
-342805164 134885145 -443725284 272797930 -169703017 -322347834 113286378 -52403961 389449025 768917995 -239280168 -320523617 485958542 -307015004 52781048 -284608826 873290646 825023359 447555558 -150604048 -213025277 558724719 618172880 268547318 296185621 137049508 -365598981 712982688 -105184809 -473999823 471468025 716735194 462695933 694760002 940174703 740344001 726389452 398715993 -131936359 737710919 19457387 326103403 270452810 493723077 490330997 -475244764 -381913882 268251612 329116102 189323843 51882054 -218754417 681718820 697969831 435106095 215677404 353047815 576698424 -409339614 699706773 646597790 49383371 -370019922 868522207 -189556931 739794931 892003744 76860185 659056325 352332597 909037637 474376180 793249150 492153020 515608906 519669994 810790383 -274101026 -394714088 822724196 357422916 154103459 -471568690 744048986 529630485 507475959 -12472545 -467757935 295202236 -34384423 815452552 315761179 295989407 665796337 548293339 693464167 359653635 822355240 -304148913 205357712 -475314977 -360461664 693522191 587769727 710143070 368887681 200209214 187363621 818819092 162174138 -418575918 530465442 128857386 -107207031 -7589576 -52970091 -126073112 965638651 243877305 -474950173 753513202 -115000451 -449880559 941744488 -285359742 647465316 -385999477 908878515 -305920375 910162418 499274548 615168899 -436435496 -409350291 -311432408 698390625 159775755 527766497 741955710 -182704534 -21569893 3949526 739318553 661841772 478609903 -196543537 -87405544 -394364236 24597007 780828951 -319838072 314913236 707590759 295206415 -90465154 100451691 -357517426 795908051
42686723 610597178 807365763 244472620 -332360496 969417737 -384768540 -282061722 393808569 67755156 -84354896 -176919149 -347897559 521965968 481813254 612406303 8227504 386184338 138807434 624723766 -398667027 278284929 -164493396 80333309 126983327 986487632 135194722 701135260 -123845772 -65333966 -338408135 591220635 -12324127 685107828 412030837 847036396 -342720590 298100801 621538379 33971760 297702191 740688763 -102135463 171518523 70267823 -427009861 69589091 769555779 740652175 -417225725 -331749032 -360592362 21552537 -155617102 114126958 219958499 265243716 753150283 455956347 173445426 557476438 390990728 753534213 -153897177 -172603400 348159024 45308631 634879571 741535322 963225142 148075636 57515462 -99209946 467959115 -246258420 506033544 28793506 858373238 400379270 858165585 -63970073 901206629 207072079 844491122 -495111551 142249988 -219357731 459770160 654287098 134902145 361065745 -330109718 506437451 -418699954 163875055 416188070 630310764 501657182 -101263473 257281512 -444912777 428622419 270070398 582788927 789105338 430271770 5019767 529004167 -334645199 157610782 410253735 619840434 430436502 289781255 -25030715 572212612 880060693 907115763 893190712 427989140 -448537554 -116836946 -190608342 760324554 -78496505 -212883926 388181376 563657820 937723028 534205541 783320566 -168132766 -286406217 738288159 -349651106 -114156196 -257186704 400987177 860724493 524598451 924619538 -249229581 -295434836 -14073937 -111988194 519987101 933858997 -286720278 549501402 739065274 104838592 409757137 -359258285 -246474719 482983119 385338080 290095067 324787668
-194908380 -131123608 145816332 938745597 693048154 -83537375 675255064 224324283 666147674 756221780 -95890814 266326256 222900650 281913693 243623916 454336403 -377156912 389644940 -342670020 480140585 992387201 726584106 529038833 598402070 981717654 -214544708 303198571 718941976 43105276 824660743 204192386 136902343 -185047503 747468293 403145378 713746778 241161730 18718177 -434100820 326844865 598446138 619971917 998057422 -171236733 345167047 876793786 -423538357 674579700 223462367 890724868 -376193850 179262619 872370490 -51377965 728314928 741389389 958756612 118164987 189678138 190082619 -328880636 -375879132 592451377 -139573099 934523994 352893801 -242908833 -147451603 458423994 39901230 692351212 228499089 408566219 627848761 75299 -493313652 -303050236 -286205366 -448403200 -438599958 255703266 909144639 36346684 614136558 609517669 154778915 -186500445 810932441 -429216675 -361414357 611758553 431483134 713148271 981518032 -77217969 741978015 178185363 -282316306 -354983061 770955483 106423509 -47155661 804819838 199954749 -334896392 -135463602 862985259 613041064 445895269 756431828 -270351329 909889845 761589098 660005769 -84532299 -306033246 623660867 -414672895 163849385 590379392 612046419 -190596977 262913234 40160565 218885469 229293210 -428173393 564458575 590768545 774012408 499872571 -191874742 780333995 756453640 -156638068 -421527048 549481062 -266943518 238914481 668018024 727612906 708165087 823404623 821578715 -3623396 539912746 850225520 -37742370 400588930 559936953 825599337 -114556921 724156344 901384236 725813181 -38414517 -414982031 336742342
185143073 -2856989 -371650063 226371620 783029062 -496814793 805941407 -62091616 29459335 448380630 673994609 345104078 -121186141 25890243 694930096 272365744 -21302566 274322012 -35620077 124544784 499997618 75888110 380338338 621672178 969764077 990076223 495392229 643124342 313143643 -244138508 34648889 543664250 -457044562 497564325 458816549 338219309 229730386 234310168 483079089 196558863 257301368 850255691 492047584 954540142 -80457142 -362212099 -366656601 954973367 -383442176 967142399 662491250 430050248 619538281 556710506 846130333 799993446 -130049621 656256688 957465740 915967349 -431306689 199075941 205008640 -430322064 180253259 570422193 895055868 -400793579 236281634 418607173 346121065 672201835 -317537845 781944569 145088232 296157674 845260086 -235257575 368769441 -92176228 245000940 92251650 63606700 36432644 829320707 120475584 -210430668 781988321 565605536 -150554701 -48267772 -482391685 -323330095 908803571 151626829 475167309 904348155 -124254673 938919618 -161830138 844395449 -322401297 607065207 633927209 294294942 -440446435 -296309315 841376061 -149587040 471900110 -295630586 972218013 177849890 89543412 707512369 -394452556 -33891831 792258607 113497790 972413163 495777976 634649625 -49358294 949019967 807669514 150075727 -238441034 375983157 -82955656 194074914 239991378 296696719 33060888 -203804834 -232466009 40068836 620121733 51562083 -432280913 -337415032 225223475 -334575111 122853097 931901594 616740199 -133783086 540320672 -21089667 -62982401 639704178 -409971364 -220393003 684629764 -364328126 -489014871 -351665017 -499242384 104873525
-118284224 955365660 443565426 -82542187 664299654 -107733396 -27720583 -106076219 -19017627 162694344 268736220 41056594 912027073 723889925 -318030202 532657029 837987272 288072323 926016048 605324651 945970046 -124022589 660439552 -499349048 388087245 442437721 196570667 -493449007 -200864551 128451677 -102760247 61756320 -483658924 -387367312 -463334879 -485784212 409862175 644980183 317004806 58024596 421300941 469757796 -488735578 957772838 356011440 507857512 6547932 987838691 88076943 76716378 390515090 -218393989 169382773 954416965 -433423460 165635211 -294071642 550140082 -407182970 13778613 342989066 -343138658 492786236 861496179 223073815 388895584 733617033 -363720549 -492271532 321875527 -103477726 49730617 531996702 -481849344 -136967498 390479143 24823957 -314221869 94342290 641164016 -250654634 -36667229 -35941465 376372519 736594423 42612488 981822830 -9996764 -23523810 841198890 911275174 977192920 -239831456 827240260 -233179186 1328044 989052503 883480195 -348435177 -234930713 2709411 -254340928 776949526 -192592489 893553621 833783421 106230414 224995244 99270644 542277943 891369970 539057785 682641778 650618209 673243262 -362432534 495909243 -473721610 567431172 -77048128 -146994357 -161300668 71530577 -33121480 -107205090 862245105 194561440 576856343 861729766 999359267 734860766 341093678 -400762265 561161615 539307864 -277838937 -411045858 -103357799 712985269 -452776889 -178220722 28412469 -272810349 925293340 463162166 886156422 949292430 -473909976 -380734159 396537570 87551851 -264018353 793999602 268864584 431464503 -254697068 680508304 -195708123
720460679 -83975037 706150794 565020395 -149169873 387565849 14096824 -140356182 985184324 -94604052 16156388 -428388369 672051566 950318817 -485906372 5995158 -100793027 162383035 29526764 952167680 -402967145 324877100 -233639364 906755083 230730272 86160152 638350531 342413407 756600023 -32884104 -141231297 194193897 963317001 -327169951 -164646313 440879386 834148049 251372148 -193293337 -402941601 886739806 -73941707 309327434 837757468 -472603109 -382821664 -328067073 714484268 -253504203 -479240853 407755179 91133466 686131989 30344101 -31155623 740270114 731110294 686845636 -422809320 999441699 520116244 117558002 918083301 943880941 855554598 -426511351 155449235 486028356 809072307 525384142 285907570 340495273 -8105215 119873001 -349922113 -461962296 -472905993 396170548 809030036 986420023 -417702609 596436465 -257653685 -415762044 -309114836 -449841241 703268387 -338155040 335110695 252124399 279006549 949039315 976576341 -197391201 -349498963 11309376 -330291885 -254267553 -66616534 250855823 -288443891 288184285 781220727 322977530 -321810257 583188707 -250261642 430787034 -187944088 551119264 -452765365 920586835 917991056 74615451 169574455 -238457411 30773524 535295532 -223476295 187468036 953176737 190626883 427482756 8195009 907058432 -286198424 353894415 542570852 285173541 527676422 123957633 155040616 -209064178 -153905370 -214787156 226614654 960870333 157527998 397831793 502127979 352492761 79605414 -154132517 -286208802 41239651 -29624888 21035887 226902307 919840256 343056906 -334308356 -174676775 125214003 -356372477 -477242182 -388667497 -331037304 481071485
-276665822 189135117 586393459 -381513971 317874145 331149735 269405971 -282043073 112558023 678840551 239581840 -11382367 116349182 838000394 -60154905 284322524 93066226 510993633 532028074 212821197 -264690116 -70816686 -336149089 -372612971 752505155 362762056 -59098894 -345888496 356568006 352177100 584303359 168118547 939030392 250372920 -497688866 267591468 -114840483 -442422161 -355941052 926245787 -56230515 -468161102 122028907 65822105 -172123705 439472035 -414295718 180899286 -434249900 471010684 905994040 153511784 403071830 390078765 900317162 840257910 279540114 988447064 -350058545 931342990 610541447 231718864 -333514472 -470891910 355573862 644261849 461456897 -365884574 133627701 989118018 21672915 -311875898 992131367 -346665733 -146561388 458143581 921177427 795244500 -100938171 -290069107 585322591 -27342654 -23824876 -466563665 -138477183 387525195 571437479 614879298 976981052 901525824 550999324 317798625 533475239 -419936098 535003338 864186306 -118883362 806511623 -486894069 389052213 331820219 -230643763 421994130 182635302 19367463 -407009619 343181860 237314726 -430528518 -30332999 254619374 -110143959 589006393 -292316362 -440960432 -273806211 -169320013 273038590 273834317 -267966345 215614256 822404066 924981893 -63433394 810675934 41996247 410777920 -488238390 693019603 928174241 29904330 361792862 486295480 488538915 228037175 919556909 756690185 692089452 800511794 562820871 -60823161 110057051 131673107 231121525 85898049 209118884 365953131 613794414 -381652035 687335870 974892404 -140703487 338212003 -405950235 -99945595 185514758 971847102 437785943
326237047 699463934 889736170 -143159686 346320489 327558129 -250943411 -46427555 -11046922 760709947 -242884862 748687806 -328335193 933354539 -491398366 -377422216 730025654 -351853127 932921415 361676167 -473558731 801774918 875235185 522021417 -492686153 17785648 -150766272 -7710168 294573422 -460190557 -357567761 248336727 468018507 125946460 -93063695 835491831 -471886754 886190872 897802335 19248122 -37664839 -52754000 -43642463 -305034272 339127388 -375826510 366209002 274877916 -163894934 616886553 -304436979 217661438 786167820 497071333 -339922119 960724082 77524279 825109307 681853413 -56545056 5194466 297715554 456058616 715249648 627962102 814744361 608347044 131823693 -49431495 -461656765 361352562 -315053455 913359302 -354294048 -65202011 221532955 -4791307 -449246073 -87033377 -429211326 -499251688 610715516 -94296483 596898352 110244058 226033581 -77118387 173050989 261470577 -339065019 -199374379 36809491 93202889 476777871 -376060027 704298581 599867954 106556157 60372228 -477849078 385198745 248694483 260942353 -452826242 -340839191 -234281944 515915381 18306936 929999362 242856606 -223875708 -234899800 573644903 144135903 -377998946 -242180556 316022125 643799438 865426762 416767731 869037487 850551273 -467958689 809038499 -270838312 303847071 25118509 -221740734 692360109 98480763 825894696 856373342 -410402 634336848 79636922 -375609126 -105619149 278759881 109127972 579930131 291088593 173957211 331280087 36125225 420313691 823346508 419161587 500290412 -130525385 494878287 -280936863 -267005209 815687942 457079321 189457965 984802774 -223657084 658688711
