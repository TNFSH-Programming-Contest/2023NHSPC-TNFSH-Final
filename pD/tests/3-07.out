1272548660889
